app/assets/javascripts/app/cart.coffee

```diff
@@ -0,0 +1,13 @@
+$ ->
+  class Cart
+    constructor: (cartSelector) ->
+      @$cart = $(cartSelector)
+      @$cart.find("[data-quantity-field]").each @setupQuantityListeners
+
+    setupQuantityListeners: (i, link) ->
+      trackedField = $(link).data("quantity-field")
+      $("##{trackedField}").change ->
+        newQuantity = $(@).val()
+        link.href = link.href.replace(/quantity=\d+&/, "quantity=#{newQuantity}&")
```

Comment:
Wondering if it would be a bit safer to do something like:

```javascript
link.href.replace(/\bquantity=\d+(&|$)/, "quantity=#{newQuantity}$1");
```

That should be able to handle `quantity` being the last param, as well as avoid accidentally replacing the value of another param that happens to end with "quantity".